docs/backend/backend-overview.md

```diff
@@ -0,0 +1,210 @@
+# Backend Service - Future Vision
+
+This document outlines a future high-level vision for what the Backend Service
+could look like as an evolution from where we are today.
+
+Goals:
+
+- Specify key requirements for an evolved high-level architecture
+
+Non Goals:
+
+- Providing a detailed description of how the current backend is implemented.
+- Getting into implementation details on a component level. We may cover key
+  technology choices, but not details on how they are to be used.
+
+## Personas
+
+The following personas will be referenced throughout the document:
+
+- **Cluster Administrator**: The person responsible for managing a Kubernetes cluster.
+  This person cares about all hardware and its associated hardware and
+  cluster-wide software resources.
+
+- **Backend Service Owner**: The person responsible for managing the Backend Service
+  on top of the cluster provided by the Cluster Administrator.
+
+- **InstructLab Model Developer**: This person is a user of the Backend Service.
+  They don't care about the underlying infrastructure details. They care about
+  data sets, model training, model evaluation, and all of those types of related
+  details associated with the pipeline implemented by the Backend Service.
+  - **Sub-category A**: A Model Developer who is trying to add domain-specific
+  knowledge to the model and is very interested in how they have improved (or
+  not) the model in that specific domain
+  - **Sub-category B**: A Model Developer who is interested in improving an
+  overall model benchmark
+
+## Guiding Principles for the Future
+
+We aim to articulate a set of guiding principles that will help guide our
+technical decision-making as we evolve the Backend Service. These principles are
+intended to be high-level and should be used to inform the design of the system
+as a whole.
+
+When faced with a difficult technical decision with a set of trade-offs, we should
+review these principles to ensure we make a decision that aligns with them.
+We should also consider these principles to be open to change as we learn more,
+but they will always serve as a guide.
+
+Summarized, these principles are:
+
+- Scalable
+- Service-Oriented
+- Observable
+- Reusable
+- Small Iterations
+- Minimize Mandatory External Dependencies
+- Make Development Accessible
+- Model Independence
+
+### Scalable
+
+The initial iteration of the Backend Service targeted a particular set of clusters
```

Comment:
Yeah -- this was referring to an older implementation, not the one proposed in this doc that would work differently.  apologies if that didn't come through correctly.

An earlier iteration of this doc had a whole lot more detail about the "old" system, but I ripped it all out since it's not really so relevant (nobody can see the details of it, anyway)